### DISCUSSION TITLE
How do you add external data (e.g. images) in class style?

### DISCUSSION THREAD

**Discussion by slussinator** [ctreffe/alfred#164] [Alfred Q&A]
I would like to insert text/images into my experiment using a path to data on my computer. In instance style I can just add it using 

```
page += al.Text(path="files/text")
page += al.Image(path="files/image")
```

is something similar possible using class style?

**Comment by Adsecularis** [comment]
Yes, it is.

```
@exp.member
class PicturePage(al.Page):

    def on_first_show(self):
        self += al.Text(path="files/text")
        self += al.Image(path="files/image")
```
Though you can use any other hook that is appropriate for your situation.
